src/CacheChecker.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Library\HealthCheckCache;
+
+use OAT\Library\HealthCheck\Checker\CheckerInterface;
+use OAT\Library\HealthCheck\Result\CheckerResult;
+use Psr\Cache\CacheException;
+use Psr\Cache\CacheItemPoolInterface;
+use Throwable;
+
+final class CacheChecker implements CheckerInterface
```

Comment:
please do not make it final, it's a vendor lib class we would have high chances to extend

Reply:
As I mentioned, I prefer composition over inheritance and final a really good help with that. But in order not to differentiate from the other health checker libraries, I'm okay with removing it.

Reply:
- a non final class does not prevent at all composition over inheritance usage :)
- providing final classes is imo when a class must not change no matter what (like implementation of a standard for ex), and it's not the case here
- dont forget our scope is to offer an easy to use library, you dont know how users will use this, therefore allowing the maximum flexibility is imo the best way to do. final usage goes against this way of thinking

thx for the fix !

Reply:
okay, if allowing maximum flexibility this way is the main point of view, I'll remove final then.